src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ControllerSaveTempDataPropertyFilter.cs

```diff
@@ -19,10 +25,29 @@ public void OnActionExecuted(ActionExecutedContext context)
         /// <inheritdoc />
         public void OnActionExecuting(ActionExecutingContext context)
         {
-            Subject = context.Controller;
             var tempData = _factory.GetTempData(context.HttpContext);
+            var viewData = context.HttpContext.Items[ViewDataDictionaryControllerPropertyActivator.ViewDataKey] as ViewDataDictionary ??
+                new ViewDataDictionary(_modelMetadataProvider, context.ModelState);
+
+            var lifeTimeContext = new PropertyLifetimeContext(tempData, viewData);
+            _lifetimeThingie.Init(context.Controller, lifeTimeContext);
+            // SetPropertyVaules(tempData, Subject);
+        }
+
+        public void OnResultExecuted(ResultExecutedContext context)
+        {
+            
+        }
+
+        public void OnResultExecuting(ResultExecutingContext context)
```

Comment:
This effectively writes `TempDataAttribute` properties as soon as the controller's result starts executing. We've had some users expect properties set via `TempData` in their controllers to show up in their Views (https://github.com/aspnet/Mvc/issues/7130).